src/Internal/Payment/State/ProcessedState.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Class ProcessedState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WC_Payments_API_Payment_Intention;
+use WCPay\Core\Exceptions\Server\Request\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Server\Request\Immutable_Parameter_Exception;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Internal\Service\OrderService;
+use WCPay\Internal\Service\PaymentRequestService;
+
+/**
+ * This state is used when payment is completed on the server and we need to update date on the plugin side.
+ */
+class ProcessedState extends AbstractPaymentState {
+
+	/**
+	 * Order service.
+	 *
+	 * @var OrderService
+	 */
+	private $order_service;
+
```

Comment:
There are some weird spacings, let's run `npm run lint:php` and remove them from the PR.

Reply:
npm run lint:php didn't linted the files, which is strange, so I manually linted each of these. Commit can be found here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/96812499ac286f9767224082ef325de6e6ff015e